integration-tests/src/tests/client/resharding.rs

```diff
@@ -1583,5 +1593,94 @@ fn test_latest_protocol_missing_chunks_high_missing_prob() {
     test_latest_protocol_missing_chunks(0.9, 27);
 }
 
+fn test_shard_layout_upgrade_error_handling_impl(
```

Comment:
I'm not sure how do we ensure this, but we should make sure we are deleting the snapshot before resharding is initiated from catchup. This means that catchup should NOT happen for the first block of the resharding epoch.
```
        // Make sure that catchup is done before the end of each epoch, but when it is done is
        // by chance. This simulates when catchup takes a long time to be done
        let should_catchup =
            self.rng.gen_bool(P_CATCHUP) || (next_height + 1) % self.epoch_length == 0;
```

Reply:
Good catch, will fix! 

Just to be technically correct, I'm not deleting the snapshot, I'm corrupting it (deleting some data from it).

Reply:
Yes, right! Sorry, I meant corrupt. But yeah, we won't get a chance to corrupt before resharding if catchup happens on first block of new epoch.